code/modules/mob/interactive.dm

```diff
@@ -376,70 +423,113 @@
 					STR.attackby(W, src)
 				else
 					STR.attack_hand(src)
-		interest = interest + 25
+		interest += targetInterestShift
 		doing = doing & ~INTERACTING
 		timeout = 0
 		TARGET = null
 	else
 		tryWalk(TARGET)
 		timeout++
-	if(!doing)
+
+	if(doing == 0)
 		interest--
 	else
 		interest++
+
 	if(inactivity_period > 0)
 		inactivity_period--
 
+	if(interest <= 0 || timeout >= 10) // facilitate boredom functions
+		TARGET = null
+		doing = 0
+		timeout = 0
+		myPath = list()
+
 	//this is boring, lets move
-	if(!doing && !isnotfunc() && !TARGET)
+	if(!doing && !IsDeadOrIncap() && !TARGET)
 		doing |= TRAVEL
 		if(nearby.len > 4)
 			//i'm crowded, time to leave
 			TARGET = pick(target_filter(urange(MAX_RANGE_FIND,src,1)))
-		else if(prob((FUZZY_CHANCE_LOW+FUZZY_CHANCE_HIGH)/2))
-			//chance to chase an item
-			TARGET = locate(/obj/item) in urange(MIN_RANGE_FIND,src,1)
-		else if(prob((FUZZY_CHANCE_LOW+FUZZY_CHANCE_HIGH)/2))
-			//chance to leave
-			TARGET = locate(/obj/machinery/door) in urange(MIN_RANGE_FIND,src,1) // this is a sort of fix for the current pathing.
 		else
-			//else, target whatever, or go to our department
-			if(prob((FUZZY_CHANCE_LOW+FUZZY_CHANCE_HIGH)/2))
-				TARGET = pick(target_filter(urange(MIN_RANGE_FIND,src,1)))
-			else
-				TARGET = safepick(get_area_turfs(job2area(myjob)))
+			var/choice = rand(1,50)
+			switch(choice)
+				if(1 to 10)
+					//chance to chase an item
+					TARGET = locate(/obj/item) in favouredObjIn(oview(MIN_RANGE_FIND,src))
+				if(11 to 21)
+					TARGET = safepick(get_area_turfs(job2area(myjob)))
+				if(22 to 41)
+					TARGET = pick(target_filter(favouredObjIn(urange(MAX_RANGE_FIND,src,1))))
+				if(42 to 50)
+					TARGET = pick(target_filter(oview(MIN_RANGE_FIND,src)))
 		tryWalk(TARGET)
 	LAST_TARGET = TARGET
 
+/mob/living/carbon/human/interactive/proc/favouredObjIn(var/list/inList)
+	var/list/outList = list()
+	for(var/i in inList)
+		for(var/path in favoured_types)
+			if(ispath(i,path))
+				outList += i
+	if(outList.len <= 0)
+		outList = inList
+	return outList
+
 /mob/living/carbon/human/interactive/proc/callfunction(Proc)
 	set waitfor = 0
 	call(src,Proc)(src)
+
 /mob/living/carbon/human/interactive/proc/tryWalk(turf/TARGET)
-	if(!isnotfunc())
+	if(!IsDeadOrIncap())
 		if(!walk2derpless(TARGET))
 			timeout++
 	else
 		timeout++
 
+/mob/living/carbon/human/interactive/proc/getGoodPath(target,var/maxtries=512)

```

Comment:
I can remove this if you want, but it's almost completely lagless compared to A* and i'd prefer to leave it in the future it can be used for short range tasks.